### PR TITLE
Fix missing listing preference ID issue #168887034

### PR DESCRIPTION
**Pull request by akegan**
https://www.pivotaltracker.com/story/show/168887034

### CODE REVIEW

File: app/models/force/lease.rb [hunk 1]
```diff
@@ -50,22 +50,5 @@ def to_domain
 
       domain_fields
     end
-
-    def self.date_to_domain(api_date)
```
Comment (akegan) [inline app/models/force/lease.rb:54]: moved to object base

File: app/javascript/utils/fieldSpecs.js [hunk 2]
```diff
@@ -15,10 +15,16 @@ const getRenderType = (value) => {
   }
 }
 
+const dateIsJson = (date) => {
+  return (typeof date === 'object' && 'month' in date && 'day' in date && 'year' in date)
+}
+
 export const formatValue = (value, type) => {
```
Comment (akegan) [inline app/javascript/utils/fieldSpecs.js:22]: this is the issue that came up in the test when I had updated the domain application to have it be an object.

**Review by akegan — COMMENTED** [review]
couple of contextual comments, let me know if you have any other questions

File: app/javascript/components/applications/application_form/preferences/PreferenceForm.js [hunk 3]
```diff
@@ -48,10 +33,33 @@ const removePreference = (form, i) => {
...
 
+const clearPreference = (form, i, target) => {
+  // If user switches to "Select One..." empty the preference state
+  if (target === '') {
+    form.change(`preferences[${i}]`, {})
+  } else {
+    // Reset the form state for preference additional options
+    // if they've been touched so they don't show errors
+    for (const field in form.getState().touched) {
+      if (field.startsWith(`preferences.${i}`)) {
```
Comment (LoganArnett) [inline app/javascript/components/applications/application_form/preferences/PreferenceForm.js:44]: Still such an interesting syntax but definitely a great find

**Review by LoganArnett — APPROVED** [review]
Working locally as expected, semaphore may need to be just re-run on the expander errors

**Comment by akegan** [comment]
tests passing now, but when writing up more thorough review instructions I realized I hadn't checked that lease start dates were still working, and they're not. Going to fix it now.

**Comment by akegan** [comment]
Lease start date issue fixed now, just needed to permit the params in the leases controller

**Pull request merged**